### PR TITLE
Stop plugin once target offset is reached.

### PR DESCRIPTION
**Pull request by gardnervickers**
Before merging, we should consider making this a more general stop predicate.

### CODE REVIEW

**Comment by lbradstreet** [comment]
I know I suggested the stop predicate, but let's implement that later and ship this.

**Comment by lbradstreet** [comment]
I know I suggested the stop predicate, but let's implement that later and ship this.

We need README.md to be updated with the new functionality.

File: src/onyx/kafka/information_model.cljc [hunk 1]
```diff
@@ -172,6 +177,7 @@
      :kafka/commit-interval
      :kafka/wrap-with-metadata?
      :kafka/start-offsets
+     :kafka/target-offset
```
Comment (MichaelDrogalis) [inline src/onyx/kafka/information_model.cljc]: Should be offsets, plural.

File: src/onyx/plugin/kafka.clj [hunk 2]
```diff
@@ -178,19 +197,30 @@
       (let [rec ^ConsumerRecord (.next ^java.util.Iterator iter)
             deserialized (some-> rec segment-fn)]
         (.set watermark (max (.get watermark) (.timestamp rec)))
-        (cond (= :done deserialized)
-              (do (set! drained true)
-                  nil)
+
+        (cond (= :done deserialized) ;; TODO: Remove this in favor of target-offsets
```
Comment (MichaelDrogalis) [inline src/onyx/plugin/kafka.clj:193]: Address now or later?

File: src/onyx/plugin/kafka.clj [hunk 3]
```diff
@@ -106,16 +106,34 @@
...
-  (.set lag-gauge 
+  (.set lag-gauge
         (reduce (fn [lag [tp offset]]
-                  (+ lag (- offset (.position consumer tp)))) 
+                  (+ lag (- offset (.position consumer tp))))
                 0
                 (.endOffsets consumer (.assignment consumer)))))
 
-(deftype KafkaReadMessages 
+(defn pause-partitions-over-target-offset
```
Comment (MichaelDrogalis) [inline src/onyx/plugin/kafka.clj]: Unused function?

File: src/onyx/plugin/kafka.clj (hunk 2 above)
Comment (lbradstreet) [reply src/onyx/plugin/kafka.clj:193]: This would require switching the tests over to remove `:done`. Up to you whether you want to do it now.

Comment (MichaelDrogalis) [reply src/onyx/plugin/kafka.clj:193]: Ok, pass for now. Circle back soon.

**Pull request merged**